### DISCUSSION TITLE
What is the best way to configure what fields we display in a association SHOW view?

### DISCUSSION THREAD

**Discussion by djalmaaraujo** [avo-hq/avo#2251] [Help]
Let's say you have a User that has_many UserAddresses. In your user resource, you have something like this:

```ruby
if view.show?
      field :user_addresses, as: :has_many, discreet_pagination: true, hide_search_input: true
    end
```

How can I configure different fields only when displaying on this User page? I want to see more fields if I go directly to the user addresses resource views, but when viewing from the User perspective, I might just need to see the Full Address + edit controls.

Is this currently possible? I could not find easily in the docs.

**Comment by adrianthedev** [comment]
We are not encouraging to use `if/else` with field visibility. This will lead to unwanted results.

Please use the [`visible` field option](https://docs.avohq.io/3.0/field-options.html#field-visibility).

In order to show/hide the field based on the fact that it's a has_one, you can check the `params`.
The `has_one` should have some extra params like the `turbo_frame` and controller to `avo/associations`
Sample below: 
```
{"turbo_frame"=>"has_one_field_show_admin", "view"=>"show", "controller"=>"avo/associations", "action"=>"show", "resource_name"=>"teams", "id"=>"3", "related_name"=>"admin", "related_id"=>"3"}
```

**Comment by adrianthedev** [comment]
CLosing to keep the queue open. Pleas feel free to reopen if needed.
